### PR TITLE
Bug fix and rework the scale-aware code

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: bug fix

KEYWORDS: scale-aware ntiedtke

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
The original PR to add scale-awareness to the new Tiedtke scheme had a bug: the introduction of the namelist control didn't work correctly. Also the introduction of the namelist control made the code look ugly. After discussing with the original developer, we agreed if the changes to scale the cloud to rain conversion coefficient (which helps to reduce very light precip), and saturation check as a condition for triggering mid-level convection (which helps to the number of grid points that are classified as mid-level convection) are removed, the rest of the code can be added without a namelist control. 

Solution:
Removing the namelist introduced in PR#1806, and the changes related to cloud-to-rain conversion coefficient and saturation check for mid-level convection, and only keeping the scale-awareness only for convective adjustment time scale and shallow mass fluxes.

LIST OF MODIFIED FILES: 
M       Registry/Registry.EM_COMMON
M       dyn_em/module_first_rk_step_part1.F
M       phys/module_cu_ntiedtke.F
M       phys/module_cumulus_driver.F

TESTS CONDUCTED: 
1. Works as expected in two cases.
2. The Jenkins tests have passed.

### CODE REVIEW

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Review by dudhia — DISMISSED** [review]

**Review by smileMchen — APPROVED** [review]

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: phys/module_cu_ntiedtke.F [hunk 1]
```diff
@@ -590,16 +584,20 @@ subroutine tiecnvn(pu,pv,pt,pqv,pqc,pqi,pqvf,ptf,poz,pzz,pomg, &
...
+      real scale_tau(lq), scale_mf(lq), relax_t_limit(lq), dxref
 !
 !  set scale-dependency factor when dx is < 15 km
 !
       dxref = 15000.
       do j=1,lq
-      if (dx(j).lt.dxref .and. ntiedtke_dx_opt.eq.1) then
-          scale_fac(j) = (1.+log(dxref/dx(j)))**3
+      if (dx(j).lt.dxref) then
+          scale_tau(j) = (1.+log(dxref/dx(j)))**3
```
Comment (hichunxi) [inline phys/module_cu_ntiedtke.F:594]: A few issues here:
1) There is inconsistency at 15 km grid spacing. scale_fac = 1. and 1.1995, respectively. 
2) There is inconsistency for relaxing time limit as well, 43200. vs. 10800.
I prefer the name scale_fac. Maybe better to use the name scale_fac2 for shallow convection.

File: phys/module_cu_ntiedtke.F [hunk 2]
```diff
@@ -1039,14 +1035,10 @@ subroutine cumastrn  &
        if(ldcum(jl).and.ktype(jl).eq.1) then
            ikb = kcbot(jl)
            ikt = kctop(jl)
-           ztau = ztauc(jl) * scale_fac(jl)
+           ztau = ztauc(jl) * scale_tau(jl)
```
Comment (hichunxi) [inline phys/module_cu_ntiedtke.F:1038]: Maybe it is better to set ztauc (360., 10800.), then ztau=ztauc*scale_fac

File: phys/module_cu_ntiedtke.F [hunk 3]
```diff
@@ -1085,10 +1077,8 @@ subroutine cumastrn  &
            else
              zmfub1(jl) = zmfub(jl)
            end if
+           zmfub1(jl) = zmfub1(jl)/scale_mf(jl)
```
Comment (hichunxi) [inline phys/module_cu_ntiedtke.F:1080]: scale-awareness for shallow convection at 15 km grid spacing is questionable.

**Comment by weiwangncar** [comment]
@hichunxi Thanks for the comments. If I change the constant in the scaling formula a bit to make the value at 15 km to be equal to that in the original formula, I get this plot:

scale

Would this be acceptable. As you can see, the green line in the plot is for scaling shallow convection, and it is a very weak scaling.

**Comment by hichunxi** [comment]
> @hichunxi Thanks for the comments. If I change the constant in the scaling formula a bit to make the value at 15 km to be equal to that in the original formula, I get this plot:
> 
> scale
> 
> Would this be acceptable. As you can see, the green line in the plot is for scaling shallow convection, and it is a very weak scaling.

For deep convection, it looks ok. But for shallow convection, the scale-factor is around 2 at 8 km grid spacing, which might be a little bit too high.

**Comment by weiwangncar** [comment]
@hichunxi You might be right. How about we leave it like this now, and make adjustment if we have evidence that it doesn't work well or is not realistic in the future.

**Comment by hichunxi** [comment]
@weiwangncar Ok, leave it as is now and make potential changes in the future.

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by weiwangncar** [comment]
@hichunxi Can you take a look at this final change? Thanks.

File: phys/module_cu_ntiedtke.F (hunk 2 above)
Comment (hichunxi) [reply phys/module_cu_ntiedtke.F]: @weiwangncar please check the code here. My suggestion is to add (360, 10800) to **ztauc**. There will be no additional time limit for ztau.

Comment (weiwangncar) [reply phys/module_cu_ntiedtke.F]: @hichunxi I thought that is what I did based on your recommendation before. Here is the section in the new code:

           ztauc(jl) = max(ztmst,ztauc(jl))
           ztauc(jl) = max(360.,ztauc(jl))
           ztauc(jl) = min(10800.,ztauc(jl))
           ztau = ztauc(jl) * scale_fac(jl)

Comment (hichunxi) [reply phys/module_cu_ntiedtke.F]: @weiwangncar Sorry I missed it. It looks good now. I will approve this PR very soon.

**Comment by weiwangncar** [comment]
@hichunxi Thanks!
@dudhia @smileMchen Can you review and approve if it is ok? Thanks!

**Review by hichunxi — APPROVED** [review]

**Review by dudhia — APPROVED** [review]

**Review by smileMchen — APPROVED** [review]

**Pull request merged**